core/src/main/java/cz/xtf/core/waiting/failfast/SupplierFailFastCheck.java

```diff
@@ -0,0 +1,45 @@
+package cz.xtf.core.waiting.failfast;
+
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Generic class for {@link FailFastCheck} so that builder can build it easily.
+ */
+public class SupplierFailFastCheck<X> implements FailFastCheck {
+
+	private final Supplier<X> eventSupplier;
+	private final Function<X, Boolean> shouldFail;
+	private final Function<X, String> reasonSupplier;
+
+	private String reason;
+
+	/**
+	 * @param supplier       supplier that provide object
+	 * @param shouldFail     condition. The result is the result of {@link FailFastCheck#shouldFail()}
+	 * @param reasonSupplier returns string as a reason why {@code shouldFail} returned {@code true}
+	 */
+	SupplierFailFastCheck(Supplier<X> supplier, Function<X, Boolean> shouldFail, Function<X, String> reasonSupplier) {
+		this.eventSupplier = supplier;
+		this.shouldFail = shouldFail;
+		this.reasonSupplier = reasonSupplier;
+	}
+
+	@Override
+	public boolean shouldFail() {
+		X resources = eventSupplier.get();
+		if (shouldFail.apply(resources)) {
+			reason = reasonSupplier.apply(resources);
```

Comment:
I'll need to see this in action. How does `reasonSupplier` looks like?

Reply:
Check this out https://github.com/xtf-cz/xtf/pull/366/files#diff-1798cb6edc311f5fdc997027b5a83e6cR86

Reply:
Ok, now I get it :-) it's a little bit different approach than I thought. In case we have number of ...`ofMessages(...)` message events based on which it failed fast then it would be great know exactly which it was. I think that then there should be added `FailFastCheck` for each event message separately with given condition to know which it was exactly, right?

Reply:
I'm not saying that it should be changed, this is probably the only way how to avoid of setting reason directly in condition. I see it as tradeoff and when creating fail fast it will need to be basically multiple failfast.

Reply:
Yeah, with current functionality it would need to be in that way (mupltiple FailFastChecks). But do we need to know which exact regex (what we are looking for) matched? The way I see it:
- we know what events show bad state
- when a bad state occurs I see list of events (in a exception) that is subset of those known events

Check this out: https://gitlab.mw.lab.eng.bos.redhat.com/jbossqe-eap/openshift-eap-tests/merge_requests/615#note_105572

Reply:
Ok, I think it's ok